### PR TITLE
TFLu: Move Ethos-U custom op out of AllOpsResolver

### PR DESCRIPTION
**Pull request by mansnils**
Addresses: https://github.com/tensorflow/tensorflow/issues/44628

### CODE REVIEW

File: tensorflow/lite/micro/micro_mutable_op_resolver.h [hunk 1]
```diff
@@ -30,6 +30,8 @@ limitations under the License.
 #include "tensorflow/lite/schema/schema_generated.h"
 
 namespace tflite {
+TfLiteRegistration* Register_ETHOSU();
+const char* GetString_ETHOSU();
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_mutable_op_resolver.h:34]: move these to a new header `micro/kernels/ethous.h`, similar to `micro/kernels/fully_connected.h`

**Review by advaitjain — CHANGES_REQUESTED** [review]
Ideally, I was hoping that we could even remove AddEthosU from AllOpsResolver to avoid going down the path of AllOpsResolver having target-specific ops. But happy to hold off on that until there are more Ops like the EthosU.

Current PR looks good except for the small nits.

File: tensorflow/lite/micro/micro_mutable_op_resolver.h (hunk 1 above)
Comment (mansnils) [reply tensorflow/lite/micro/micro_mutable_op_resolver.h]: Done

**Review by advaitjain — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

**Comment by gbaned** [comment]
@mansnils  Can you please resolve conflicts? Thanks!

**Review by advaitjain — APPROVED** [review]

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.

**Pull request merged**